packages/lesswrong/components/recommendations/RecommendationsAndCurated.tsx

```diff
@@ -143,86 +174,121 @@ const RecommendationsAndCurated = ({
-              <Link to={"/library"}>
-                <SectionSubtitle className={classNames(classes.subtitle, classes.continueReading)}>
-                  Continue Reading
-                </SectionSubtitle>
-              </Link>
-            </LWTooltip>
-            <ContinueReadingList continueReading={continueReading} />
-          </AnalyticsContext>
-        </div>}
-
-        {renderBookmarks && <div className={classes.subsection}>
-          <AnalyticsContext pageSubSectionContext="frontpageBookmarksList" listContext={"frontpageBookmarksList"} capturePostItemOnMount>
-            <LWTooltip placement="top-start" title={bookmarksTooltip}>
-              <Link to={"/bookmarks"}>
-                <SectionSubtitle>
-                  Bookmarks
-                </SectionSubtitle>
-              </Link>
-            </LWTooltip>
-            <BookmarksList limit={bookmarksLimit} hideLoadMore={true}/>
-          </AnalyticsContext>
-        </div>}
+        {renderContinueReading && (
+          <div className={currentUser ? classes.subsection : null}>
+            <AnalyticsContext pageSubSectionContext="continueReading">
+              <LWTooltip placement="top-start" title={continueReadingTooltip}>
+                <Link to={"/library"}>
+                  <SectionSubtitle className={classNames(classes.subtitle, classes.continueReading)}>
+                    Continue Reading
+                  </SectionSubtitle>
+                </Link>
+              </LWTooltip>
+              <ContinueReadingList continueReading={continueReading} />
+            </AnalyticsContext>
+          </div>
+        )}
+
+        {renderBookmarks && (
+          <div className={classes.subsection}>
+            <AnalyticsContext
+              pageSubSectionContext="frontpageBookmarksList"
+              listContext={"frontpageBookmarksList"}
+              capturePostItemOnMount
+            >
+              <LWTooltip placement="top-start" title={bookmarksTooltip}>
+                <Link to={"/bookmarks"}>
+                  <SectionSubtitle>Bookmarks</SectionSubtitle>
+                </Link>
+              </LWTooltip>
+              <BookmarksList limit={bookmarksLimit} hideLoadMore={true} />
+            </AnalyticsContext>
+          </div>
+        )}
 
         {/* disabled except during review */}
         {/* <AnalyticsContext pageSectionContext="LessWrong 2018 Review">
           <FrontpageVotingPhase settings={frontpageRecommendationSettings} />
         </AnalyticsContext> */}
+      </>
+    );
+
+    return <SingleColumnSection className={classes.section}>
+      <AnalyticsContext pageSectionContext="recommendations">
+        {titleNode}
+        {showSettings &&
+          <RecommendationsAlgorithmPicker
+            configName={configName}
+            settings={frontpageRecommendationSettings}
+            onChange={(newSettings) => setSettings(newSettings)}
+          /> }
+        {(sectionExpanded || !isEAForum) && bodyNode}
```

Comment:
Most of this diff is just moving the main body into a `bodyNode` element so that it can be toggled more easily